### PR TITLE
[Concurrency] Remove C++ runtime references from embedded Concurrency.

### PR DESCRIPTION
**Pull request by al45tair**
In embedded mode, we mustn't have references to the C++ library, because some embedded platforms don't include the C++ library.

Additionally, it's good to avoid use of global operator new and operator delete, because they can be globally overridden and this has bitten us in the past.

rdar://137286187

### CODE REVIEW

**Comment by al45tair** [comment]
@swift-ci Please smoke test

File: stdlib/public/Concurrency/TaskGroup.cpp [hunk 1]
```diff
@@ -579,7 +590,13 @@ struct TaskGroupStatus {
     swift_asprintf(
         &message,
         "error: %sTaskGroup: detected pending task count overflow, in task group %p! Status: %s",
-        group->isDiscardingResults() ? "Discarding" : "", group, status.to_string(group).c_str());
+        group->isDiscardingResults() ? "Discarding" : "", group,
+#if !SWIFT_CONCURRENCY_EMBEDDED
+        status.to_string(group).c_str()
+#else
+        "<unavailable in embedded>"
```
Comment (rjmccall) [inline stdlib/public/Concurrency/TaskGroup.cpp:597]:
```suggestion
        "<status unavailable in embedded>"
```

File: stdlib/public/Concurrency/TaskGroup.cpp [hunk 2]
```diff
@@ -822,6 +844,8 @@ class AccumulatingTaskGroup: public TaskGroupBase {
 class DiscardingTaskGroup: public TaskGroupBase {
   friend class ::swift::AsyncTask;
 
+private:
+
```
Comment (rjmccall) [inline stdlib/public/Concurrency/TaskGroup.cpp:848]: Spurious

**Review by rjmccall — APPROVED** [review]
A couple minor requests, but otherwise LGTM

**Review by ktoso — APPROVED** [review]
Lgtm

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Comment by al45tair** [comment]
Looks like some C++ problems to do with the allocator on both Linux and Windows. Will go figure out what changes I need there.

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Pull request merged**